### PR TITLE
Store TrackExtra, TrackingRecHits, and SiPixel/StripClusters for muons in AOD+MINIAOD (10_6_X)

### PR DESCRIPTION
**Pull request by bendavid**
Backport of https://github.com/cms-sw/cmssw/pull/31217

The extra changes to PATMuonSlimmer are just code formatting which are added as a separate commit at the beginning.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: DataFormats/PatCandidates/src/Muon.cc [hunk 1]
```diff
@@ -380,6 +380,47 @@ void Muon::embedCombinedMuon() {
...
 
+void Muon::rekeyEmbeddedTracks(std::vector<edm::Handle<edm::Association<reco::TrackExtraCollection>>> const& assocs) {
+  auto rekey = [&assocs](reco::Track& track) {
+    reco::TrackExtraRef const& trackExtra = track.extra();
+    for (auto const& assoc : assocs) {
+      if (!assoc->contains(trackExtra.id())) {
+        continue;
+      }
+      reco::TrackExtraRef const& trackExtraOut = (*assoc)[trackExtra];
+      if (trackExtraOut.isNonnull()) {
```
[the remark this answers is not in the document]
Comment (bendavid) [reply DataFormats/PatCandidates/src/Muon.cc:391]: Sorry, missed part of a commit when rebasing, will fix.

[the remark this answers is not in the document]
Comment (bendavid) [reply DataFormats/PatCandidates/src/Muon.cc:392]: Should be fixed now

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
+1
- Identical backport of #31217 (which also includes the harmless backport of some code-format update)
- TrackExtras are added to the AOD and miniAOD outputs, without modifications to the other objects
- Protection has been added (via modifier that selects the affected campaigns) to avoid association with TrackExtras when they are not stored in the input AOD, as in the master version of this PR
- Jenkins tests pass: additional products are there, with no difference in other reco outputs
- Content validated in the (identical) master version of this PR, see https://github.com/cms-sw/cmssw/pull/31217#issuecomment-685583008

**Comment by qliphy** [comment]
kind reminder @cms-sw/analysis-l2 @cms-sw/pdmv-l2 @cms-sw/simulation-l2 @cms-sw/upgrade-l2

**Comment by chayanit** [comment]
+1

**Comment by silviodonato** [comment]
+operations

**Comment by kpedro88** [comment]
+upgrade

**Comment by civanch** [comment]
+1

**Comment by silviodonato** [comment]
merge

**Comment by santocch** [comment]
+1

**Pull request merged**